src/lib.rs

```diff
@@ -10,16 +10,47 @@
 
 //! Support for matching file paths against Unix shell style patterns.
 //!
-//! The `glob` and `glob_with` functions, in concert with the `Paths`
-//! type, allow querying the filesystem for all files that match a particular
-//! pattern - just like the libc `glob` function (for an example see the `glob`
-//! documentation). The methods on the `Pattern` type provide functionality
-//! for checking if individual paths match a particular pattern, in a similar
-//! manner to the libc `fnmatch` function.
+//! The `glob` and `glob_with` functions allow querying the filesystem for all files
+//! that match a particular pattern (similar to the libc `glob` function). The methods on the
+//! `Pattern` type provide functionality for checking if individual paths match a
+//! particular pattern (similar to the libc `fnmatch` function).
 //!
 //! For consistency across platforms, and for Windows support, this module
 //! is implemented entirely in Rust rather than deferring to the libc
 //! `glob`/`fnmatch` functions.
+//!
+//! # Examples
+//!
+//! To print all jpg files in `/media/` and all of its subdirectories:
+//!
+//! ```rust
```

Comment:
I guess there are no files in `/media/` on Travis CI where the doc tests will run, so you might want to make this `rust,no_run` (cf. [the book](http://doc.rust-lang.org/nightly/book/documentation.html#running-documentation-tests) for more information). Otherwise, it probably won't hurt since you just iterate through zero entries.